lib/Doctrine/DBAL/Migrations/Tools/Console/Command/VersionCommand.php

```diff
@@ -85,61 +92,81 @@ public function execute(InputInterface $input, OutputInterface $output)
     {
         $this->configuration = $this->getMigrationConfiguration($input, $output);
 
-        if ( ! $input->getOption('add') && ! $input->getOption('delete')) {
+        if (!$input->getOption('add') && !$input->getOption('delete')) {
             throw new \InvalidArgumentException('You must specify whether you want to --add or --delete the specified version.');
         }
 
-        $this->markMigrated = (boolean) $input->getOption('add');
+        $this->markMigrated = (boolean)$input->getOption('add');
 
         if ($input->isInteractive()) {
             $confirmation = $this->getHelper('dialog')->askConfirmation($output, '<question>WARNING! You are about to add, delete or synchronize migration versions from the version table that could result in data lost. Are you sure you wish to continue? (y/n)</question>', false);
             if ($confirmation) {
-                $this->markAllAvailableVersions($input);
+                $this->markVersions($input);
             } else {
                 $output->writeln('<error>Migration cancelled!</error>');
             }
         } else {
-            $this->markAllAvailableVersions($input);
+            $this->markVersions($input);
         }
 
     }
 
-    private function markAllAvailableVersions(InputInterface $input)
+    private function markVersions(InputInterface $input)
     {
-        $version = $input->getArgument('version');
+        $affectedVersion = $input->getArgument('version');
 
-        if ($input->getOption('all') === true) {
-            $availableVersions = $this->configuration->getAvailableVersions();
-            foreach ($availableVersions as $version) {
-                $this->mark($version, true);
-            }
-        } else {
-            $this->mark($version);
+        $allOption = $input->getOption('all');
+        $rangeOption = $input->getOption('range');
+
+        if ($allOption && $rangeOption) {
+            throw new \InvalidArgumentException('Options --all and --range both used. You should use only one of them.');
+        }
+
+        switch (true) {
```

Comment:
I use `switch(true)` instead of
```
if (...) {
    ...
} elseif (...) {
    ...
} else {
    ...
}
```

For example like here: https://github.com/symfony/symfony/blob/a8e2c743c23fe17d360868868ec1fb35529040c9/src/Symfony/Component/Console/Helper/SymfonyQuestionHelper.php#L58

This makes the code easy to read